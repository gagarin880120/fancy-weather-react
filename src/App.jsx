import React from 'react';
import { hot } from 'react-hot-loader/root';
import styles from './App.module.css';
import NavBar from './components/NavBar/NavBar';
import InfoContainer from './components/Info';

function App() {
  return (
    <div className={styles.wrapper}>
      <NavBar />
      <InfoContainer />
    </div>
  );
}

export default hot(App);
