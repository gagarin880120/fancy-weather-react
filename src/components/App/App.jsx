import React from 'react';
import { hot } from 'react-hot-loader/root';
import PropTypes from 'prop-types';
import styles from './App.module.css';
import NavBar from '../NavBar/NavBar';
import InfoContainer from '../Info';
import MapBoxContainer from '../MapBox';

function App({ backgroundImageURL }) {
  const style = { backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.3) 0%, rgba(0, 0, 0, 0.35) 100%), url(${backgroundImageURL})` };
  return (
    <div className={styles.wrapper} style={style}>
      <NavBar />
      <InfoContainer />
      <MapBoxContainer />
    </div>
  );
}

App.propTypes = {
  backgroundImageURL: PropTypes.string,
};

App.defaultProps = {
  backgroundImageURL: '',
};

export default hot(App);
