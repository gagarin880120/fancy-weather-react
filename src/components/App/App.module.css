@font-face {
  font-family: Nunito;
  src: url('../../assets/fonts/Nunito-Regular.ttf');
}

.wrapper {
  color: #fff;
  padding: 0 5% 3%;
  height: 100%;
  background-size: cover;
  background-position: center;
  font-family: Nunito;
}
