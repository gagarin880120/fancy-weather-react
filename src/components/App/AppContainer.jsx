import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import App from './App';
import { selectBackgroundImageURL } from '../../redux/selectors';

export function AppContainer({ backgroundImageURL }) {
  console.log(backgroundImageURL);
  return (
    <App
      backgroundImageURL={backgroundImageURL}
    />
  );
}

export const mapStateToProps = (state) => ({
  backgroundImageURL: selectBackgroundImageURL(state),
});

AppContainer.propTypes = {
  backgroundImageURL: PropTypes.string,
};

AppContainer.defaultProps = {
  backgroundImageURL: '',
};

export default connect(mapStateToProps)(AppContainer);
