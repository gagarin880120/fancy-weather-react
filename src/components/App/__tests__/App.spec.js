import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create, act } from 'react-test-renderer';
import App from '../App';

jest.mock('mapbox-gl', () => ({
  mapboxgl: null,
}));

describe('App', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<App />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });

    test('should render with other elements depends on props', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<App currentWeather={{}} currentDate={{}} weeklyWeather={{}} isModalOpen={true} />);
      const result = renderer.getRenderOutput();
      expect(result).toMatchSnapshot();
    });

    test('should call onAppLoad when is mounted', () => {
      let root;
      const onAppLoad = jest.fn();

      act(() => {
        root = create(
          <App onAppLoad={onAppLoad} />
        )
      });

      expect(onAppLoad).toHaveBeenCalled();
    });
  });
});
