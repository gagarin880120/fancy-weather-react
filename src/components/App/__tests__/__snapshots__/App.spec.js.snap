// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render with other elements depends on props 1`] = `
<React.Fragment>
  <div
    className="wrapper"
    style={
      Object {
        "backgroundImage": "linear-gradient(0deg, rgba(0,0,0,.4) 0%, rgba(0,0,0,.5) 100%), url()",
      }
    }
  >
    <NavBar />
    <Info />
  </div>
  <Memo(Connect(ModalContainer)) />
</React.Fragment>
`;

exports[`App component should render without crashing 1`] = `
<React.Fragment>
  <Spinner />
</React.Fragment>
`;
